tests/blackbox_test.go

```diff
@@ -311,3 +338,25 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
 	}
 	return nil
 }
+
+//Identify and replace $uuid<num> with correct UUID
+//Variables with matching <num> should have identical UUIDs
+func replaceUUIDVars(str string, UUIDmap map[int]string) string {
+	finalStr := str
+
+	tokenizedArr := regexp.MustCompile("[$][u][u][i][d][0-9]+").FindAllStringSubmatchIndex(str, -1)
```

Comment:
You can improve the regexp which will also allow you to clean up the code a bit:
```golang
pattern := regexp.MustCompile("\\$uuid([0-9]+)")
for _, match := range pattern.FindAllStringSubmatch(str, -1) {
    num_int, _ := strconv.Atoi(match[1])
    finalStr = strings.Replace(finalStr, match[0], getUUID(num_int, UUIDmap), 1)
}
```

Reply:
Totally missed this function when reading the doc. Nice check!!